106-auto-rebalance-on-imbalanced-clusters.md

```diff
@@ -0,0 +1,524 @@
+Detector classes have different mechanisms to detect their corresponding anomalies.
+For example, `KafkaBrokerFailureDetector` utilises Kafka Metadata API whereas `DiskFailureDetector` and `TopicAnomalyDetector` utilises Kafka Admin API.
+Furthermore, `MetricAnomalyDetector` use metrics and `GoalViolationDetector` uses the load distribution to detect their anomalies.
+The detected anomalies can be of various types:
+* Goal Violation - This happens if certain [optimization goals](https://strimzi.io/docs/operators/in-development/deploying#optimization_goals) are violated (e.g. DiskUsageDistributionGoal etc.). These goals can be configured through the `self.healing.goals` option in Cruise Control configuration.  However, this option is forbidden in the `spec.cruiseControl.config` section of the `Kafka` CR.
+* Topic Anomaly - Where one or more topics in cluster violates user-defined properties (e.g. some partitions are too large in disk).
+* Broker Failure - This happens when a non-empty broker crashes or leaves a cluster for a long time.
+* Disk Failure - This failure happens if one of the non-empty disks fails (related to a Kafka Cluster with JBOD disks).
+* Metric anomaly - This failure happens if metrics collected by Cruise Control have some anomaly in their value (e.g. a sudden rise in the log flush time metrics).
+
+The detected anomalies are inserted into a priority queue where comparator is based upon the priority value.
+The smaller the priority value is, the higher priority the anomaly type has.
+
+The anomaly detector manager calls the notifier to get an action regarding whether the anomaly should be fixed, delayed, or ignored.
+If the action is `FIX`, then the anomaly detector manager calls the classes that are required to resolve the anomaly.
+
+Anomaly detection also has various [configurations](https://github.com/linkedin/cruise-control/wiki/Configurations#anomalydetector-configurations), such as the detection interval and the anomaly notifier class, which can affect the performance of the Cruise Control server and the latency of the anomaly detection.
+
+#### Notifiers in Cruise Control
+
+Whenever anomalies are detected, Cruise Control provides the ability to notify the user regarding the detected anomalies using optional notifier classes.
+The notification sent by these classes increases the visibility of the operations that are taken by Cruise Control.
+The notifier class used by Cruise Control is configurable and custom notifiers can be used by setting the `anomaly.notifier.class` property.
+The notifier class returns the `action` that is going to be taken on the flagged anomaly.
+These actions have three types:
+* `FIX` - Start the anomaly fix
+* `CHECK` - Delay the anomaly fix
+* `IGNORE` - Ignore the anomaly fix
+
+The default `NoopNotifer` always sets the notifier action as `IGNORE`, which  means that the detected anomaly will be silently ignored and no notification is sent to the user.
```

Comment:
This sentence can be confusing.
The fact that the notifier returns `IGNORE` doesn't mean that no notification is sent to the user. At least this is how I read the sentence.
We should make it clearer, stating that:

* this notifier returns `IGNORE` which means that the anomaly detector manager won't use any class to fix the anomaly.
* this notifier doesn't implement any notification mechanism to the users

I think it's important because a notifier could return `IGNORE` but at the same time notifying the user, which is NOT what the `NoopNotifer` does.